### DISCUSSION TITLE
Interpolating discrete functions on non-matching meshes

### DISCUSSION THREAD

**Discussion by massimiliano-leoni** [feelpp/feelpp#2036] [Q&A] [answered]
Hello! I am trying to assess Feel++'s capabilities to interpolate discrete functions defined on non-matching meshes, so any two meshes whose intersection is not empty.
Going through the code I found an implementation at `feelpp/feel/feeldiscr/interpolate.hpp:101`, which says

> Given a space `space` using a lagrange basis, compute the interpolation `interp` of `f` belonging to another function space.

so this seems to be able to interpolate from a Lagrange function [continuous? Discontinuous?] to other types of finite elements. Is this correct?
What else is available? Can one interpolate from, say, a different element type than a Lagrange element? Or from a DG2 to a P5 function?

Thanks in advance,
Massimiliano

**Comment by prudhomm** [comment, marked as answer]
@massimiliano-leoni you can definitely do this kind of thing:
```c++
auto Xhc=Pch<5>(mesh);
auto Xhd=Pdh<2>(mesh);
auto I = opInterpolation(_domain=Xhd, _image=Xhc, _range=markedelements(Xhc->mesh(),"some markers"));
```

there are a lot of examples in the testsuite.

**Reply by massimiliano-leoni** [reply]
Thanks! That's exactly what I was looking for :)
